shared/x509utils/names.go

```diff
@@ -91,3 +91,22 @@ func removeZone(name string) string {
 	}
 	return name[:idx]
 }
+
+// NameAsIP prepares a sanitised IP address name for matching certificates
+func NameAsIP(name string) (string, bool) {
+	if addr, err := core.ParseAddr(name); err != nil {
+		s := fmt.Sprintf("[%s]", addr)
+		return s, true
+	}
+	return "", false
+}
+
+// NameAsSuffix prepares a santised hostname for matching
+// certificate patterns
+func NameAsSuffix(name string) (string, bool) {
```

Comment:
This whole function looks a bit fishy to me.

Reply:
I thought about using a list of TLDs for safety, but this is to match against an already sanitised store so it should be harmless @karasz